src/middlewared/middlewared/plugins/filesystem_/perm_check.py

```diff
@@ -1,10 +1,32 @@
 import errno
+import os
 import pathlib
 
+from multiprocessing import Pipe, Process
+from multiprocessing.connection import Connection
+
 from middlewared.schema import accepts, Bool, Dict, returns, Str
 from middlewared.service import CallError, Service
 
-from middlewared.utils.osc import run_command_with_user_context
+from middlewared.utils.osc import set_user_context
+
+
+def check_perms(user: str, path: str, check_perms: dict, pipe: Connection) -> None:
+    set_user_context(user)
+
+    flag = False
+    for perm, check_flag in filter(
+        lambda v: v[0] is not None, (
+            (check_perms['read'], os.R_OK),
+            (check_perms['write'], os.W_OK),
+            (check_perms['execute'], os.X_OK),
+        )
+    ):
+        perm_check = os.access(path, check_flag)
```

Comment:
Do we need to wrap this in a try: except so that on any crash this side of the pipe is closed and we don't leave a dangling FD? Or maybe move the closing of the pipe to the caller?

Reply:
Definitely, thanks for pointing it out